rasa/nlu/model.py

```diff
@@ -307,25 +313,63 @@ def load(
             model_dir: The path of the model to load
             component_builder: The
                 :class:`rasa.nlu.components.ComponentBuilder` to use.
+            new_config: Optional new config to use for the new epochs.
+            finetuning_epoch_fraction: Value to multiply all epochs by.
 
         Returns:
             An interpreter that uses the loaded model.
         """
-
         model_metadata = Metadata.load(model_dir)
 
+        if new_config:
+            Interpreter._update_epochs_from_new_config(
+                model_metadata, new_config, finetuning_epoch_fraction
+            )
+
         Interpreter.ensure_model_compatibility(model_metadata)
-        return Interpreter.create(model_metadata, component_builder, skip_validation)
+        return Interpreter.create(
+            model_metadata,
+            component_builder,
+            skip_validation,
+            should_finetune=new_config is not None,
+        )
+
+    @staticmethod
+    def _update_epochs_from_new_config(
+        model_metadata: Metadata,
+        new_config: Optional[Dict] = None,
+        finetuning_epoch_fraction: float = 1.0,
+    ):
+        for p1, p2 in zip(model_metadata.metadata["pipeline"], new_config["pipeline"]):
+            if not p1.get("name") == p2.get("name"):
+                raise InvalidModelError("Inconsistent config for model to fine-tune.")
+            if "epochs" in p1:
```

Comment:
Optional: Would it be more robust if we defined a list containing names of all components which have `epochs` as a parameter and use that for comparison here? Or would it add a dependency which people might forget to update. In that case, the list can also be auto-populated by looking at the defaults of each component.

Reply:
What exactly do you think would be more robust then?